cranelift/codegen/src/machinst/buffer.rs

```diff
@@ -411,6 +414,21 @@ pub enum StackMapExtent {
     StartedAtOffset(CodeOffset),
 }
 
+/// An open patch region represents the beginning of an editable region
```

Comment:
Can we add a bit more doc here? "Represents an editable region that is currently being emitted in the `MachBuffer`. When closed, this becomes a `PatchRegion` and allows editing the machine code post-emission." or something like that?